File: frontend/test_local_proxy.py
import urllib.request

try:
    req = urllib.request.Request("http://localhost:3000/api/image-proxy?name=Cannon")
    with urllib.request.urlopen(req) as res:
        print("Cannon status:", res.status)
        print("Content-Type:", res.getheader('Content-Type'))
except Exception as e:
    print("Cannon Error:", e)

try:
    req = urllib.request.Request("http://localhost:3000/api/image-proxy?name=Archer%20Tower")
    with urllib.request.urlopen(req) as res:
        print("Archer Tower status:", res.status)
        print("Content-Type:", res.getheader('Content-Type'))
except Exception as e:
    print("Archer Tower Error:", e)

File: frontend/test_local_proxy.js
const http = require('http');

http.get('http://localhost:3000/api/image-proxy?name=Cannon', (res) => {
  console.log(`Cannon status: ${res.statusCode}`);
  console.log(`Content-Type: ${res.headers['content-type']}`);
});

http.get('http://localhost:3000/api/image-proxy?name=Archer Tower', (res) => {
  console.log(`Archer Tower status: ${res.statusCode}`);
  console.log(`Content-Type: ${res.headers['content-type']}`);
});
